pkg/streamingpromql/operators/function_over_range_vector.go

```diff
@@ -73,193 +75,35 @@ func (m *FunctionOverRangeVector) NextSeries(ctx context.Context) (types.Instant
 			return types.InstantVectorSeriesData{}, err
 		}
 
-		err = m.computeNextStep(&data, step)
+		hasFloat, f, h, err = m.RangeVectorStepFunc(step, m.rangeSeconds, m.floatBuffer, m.histogramBuffer)
```

Comment:
[nit] Couldn't we declare and initialise these here?

```suggestion
		hasFloat, f, h, err := m.RangeVectorStepFunc(step, m.rangeSeconds, m.floatBuffer, m.histogramBuffer)
```

Reply:
I was concerned about re-assigning it each loop iteration instead of re-using the existing variables. However, I think they'll only be on the stack not in the heap so it should be okay. I'll change it to your suggestion, but is my assumption here correct?

Reply:
Yep, you're correct - these should only be on the stack (although the `FloatHistogram` instance that `h` points to will be on the heap, but that's expected and unavoidable). So there's no harm in using `:=` here as far as I know.